src/NHibernate.Test/NHSpecificTest/GH948/FixtureByCode.cs

```diff
@@ -0,0 +1,386 @@
+﻿using NHibernate.Cfg.MappingSchema;
+using NHibernate.Criterion;
+using NHibernate.Mapping.ByCode;
+using NHibernate.SqlCommand;
+using NHibernate.Transform;
+using NUnit.Framework;
+
+namespace NHibernate.Test.NHSpecificTest.GH948
```

Comment:
In fact we favor putting tests in their functional area, so it would be even better to move them to Criteria/ProjectionsTest.cs.

Reply:
> I didn't do it because I have my set of entities specifically for my needs and I don't like hbm configuration (and it's not clear to me how to mix them).

Better add another test file in such case, like `Criteria/EntityProjectionsTest.cs`. Mixing mapping methods is not an option.

> It doesn't look to me as clean solution to mix all sort of various tests depending on different data in one file and do set up/clean up for all of them for each test method. I would really prefer keep my tests in separate class

I do not really get that point since that is the fate of most test fixtures, even your isolated current one: it already sets up a composite entity for the last tests, unused by the other tests.
Of course we should try to reuse other test entities if possible by default. But when there is a good reason, not re-using them is ok too. Anyway when it is impractical to add to the existing fixture, you can add just another one in the same folder as I propose just above.

> Maybe it's better to introduce Criteria/Projections folder? And put files there (for now two files - ProjectionsTest.cs and new EntityProjectionTest.cs)

They do not need their own folder. There are not so many fixtures in Criteria.